chapters/common_pitfalls.md

```diff
@@ -0,0 +1,43 @@
+
+### Multiple pipelines
+
+A `VkPipeline` contains the combination of state needed to perform a draw call. For every input combination (shaders, vertex layout, primitive assembly, depth testing, blending mode, etc), a new pipeline is needed. This necessitates creating and binding many pipelines for complex rendering situations. While it might appear beneficial to try to create as few pipeline as possible by increasing complexity elsewhere, it is best to profile and determine if this is truly a big performance penalty in the first place.
+
+### Multiple queues per queue family
+
+Several hardware platforms have more than one `VkQueue` per queue family. This can be useful by being able to submit work to the same queue family from seperate queues. While there can be advantages, it isn't necessarily better to create or use the extra queues. For performance reccomendations, refer to hardware makers best practices guides.
+
+### Descriptor Sets
+
+Descriptor Sets are designed to group data together by their usage and update frequency. It is reasonable to have a descriptor set bound for each of group of resources needed in a shader, for example object matrices, lights, textures, material information, and per-instance data. Vulkan requires hardware to be able to support at least 4 concurrently bound descriptor sets, as described by the maxBoundDescriptorSets value in `VkPhysicalDeviceLimits`, with many supporting 8 or more. Therefore is safe to use more than 1 descriptor set at a time. While there may be situations that may make a single descriptor set advantageous, refer to a best practices guide for specific details.
+
+### Resource duplication per swapchain image
+
+A common technique to increase throughput is to make multiple instances of a resource, where each instance can be used while rendering a different frame. The swapchain is one such example, as it contains multiple images. This can lead to the assumption that the number of instances each resource has should be the same as the swapchain image count. The issue with this assumption is that very few resources need to be duplicated the same amount as the swapchain image count. While it useful for some things, for example the number of semaphores used to signal when a swapchain image is available, it is worth understanding which resources have one to one relationships and which don't.
+
+### Vulkan is a box of tools
+
+In Vulkan, there are often multiple solutions to a problem. Consider passing data to a shader. One can use uniform buffers, storage buffers, vertex attributes, instance buffer, push constants, and more. Are all valid methods each with their own benefits and drawbacks. There is rarely a "perfect" solution to any problem. Therefore, use the tools available and create a good solution instead of trying to find a perfect solution.
```

Comment:
Consider passing data to a shader, one

Reply:
These are all valid methods...

Reply:
I feel this section needs rewriting. It lacks a clear description of what is being stated. 
My goal is to describe that Vulkan offers a lot of tools, that they are sometimes suited for a particular job, but there isn't always a specific tool for every job and developers will need to create a solution given the tools rather than rely on the API to provide the exact solution.

Reply:
I really liked the `There is rarely a "perfect" solution to any problem.` statement and why I wanted this to be 2nd on the list as its a good overall "warning"